homeassistant/components/unifi/sensor.py

```diff
@@ -185,3 +236,172 @@ async def options_updated(self) -> None:
         """Config entry options are updated, remove entity if option is disabled."""
         if not self.controller.option_allow_uptime_sensors:
             await self.remove_item({self.client.mac})
+
+
+@dataclass
+class UniFiDeviceSensorEntityDescription(SensorEntityDescription):
+    """Describes UniFi device sensor entities."""
+
+    enabled: str | None = None
+    value_fn: Callable[[UniFiDevice], Any] | None = None
+
+    def get_value(self, device: UniFiDevice) -> Any:
+        """Return value from UniFi device."""
+        if self.value_fn is not None:
+            return self.value_fn(device)
+
+        return None
+
+    def is_enabled(self, device: UniFiDevice) -> bool:
+        """Return whether the entity should be enabled."""
+        if self.enabled is None:
+            return True
+
+        return device.raw.get(self.enabled, False)
+
+
+DEVICE_SENSORS: tuple[UniFiDeviceSensorEntityDescription, ...] = (
+    UniFiDeviceSensorEntityDescription(
+        key=CPU_UTILIZATION_SENSOR,
+        name="CPU Utilization",
+        native_unit_of_measurement=PERCENTAGE,
+        icon="mdi:speedometer",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        enabled="system-stats",
+        value_fn=lambda device: float(device.raw.get("system-stats", {}).get("cpu")),
+    ),
+    UniFiDeviceSensorEntityDescription(
+        key=MEMORY_UTILIZATION_SENSOR,
+        name="Memory Utilization",
+        native_unit_of_measurement=PERCENTAGE,
+        icon="mdi:memory",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        enabled="system-stats",
+        value_fn=lambda device: float(device.raw.get("system-stats", {}).get("mem")),
+    ),
+    UniFiDeviceSensorEntityDescription(
+        key=UPTIME_SENSOR,
+        name="Uptime",
+        icon="mdi:clock",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda device: dt_util.now() - timedelta(seconds=device.raw["uptime"]),
+    ),
+    UniFiDeviceSensorEntityDescription(
+        key=TEMPERATURE_SENSOR,
+        name="Temperature",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        enabled="has_temperature",
+        value_fn=lambda device: device.raw["general_temperature"],
+    ),
+    UniFiDeviceSensorEntityDescription(
+        key=CPU_TEMPERATURE_SENSOR,
+        name="CPU Temperature",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        enabled="temperatures",
+        value_fn=lambda device: next(
```

Comment:
Multi-line lambdas aren't allowed

Reply:
What do you mean by not allowed? It was `black` that made this `lambda` multi-line...

Reply:
Code structure wise its not allowed.

See comments here https://github.com/home-assistant/core/pull/70161